src/context/AuthContext.js

```diff
@@ -1,35 +1,119 @@
-import React, { createContext, useState } from 'react';
-import { useNavigate } from 'react-router-dom';
+import React, {createContext, useState, useEffect} from 'react';
+import {useNavigate} from 'react-router-dom';
+import axios from "axios";
+
+//zie https://www.npmjs.com/package/jwt-decode
+import jwt_decode from "jwt-decode";
 
 export const AuthContext = createContext({});
 
-function AuthContextProvider({ children }) {
-  const [isAuth, toggleIsAuth] = useState(false);
-  const navigate = useNavigate();
-
-  function login() {
-    console.log('Gebruiker is ingelogd!');
-    toggleIsAuth(true);
-    navigate('/profile');
-  }
-
-  function logout() {
-    console.log('Gebruiker is uitgelogd!');
-    toggleIsAuth(false);
-    navigate('/');
-  }
-
-  const contextData = {
-    isAuth: isAuth,
-    login: login,
-    logout: logout,
-  };
-
-  return (
-    <AuthContext.Provider value={contextData}>
-      {children}
-    </AuthContext.Provider>
-  );
+function AuthContextProvider({children}) {
+    const [Auth, setAuth] = useState({
```

Comment:
Variabelen altijd camelCase schrijven
```suggestion
    const [auth, setAuth] = useState({
```

Anders raak je in de war met componenten, want die schrijf je wél PascalCase.